### ISSUE TITLE
See all validation errors at once with something liike assertsSoftly { ... }

### ISSUE THREAD

**Issue by jmfayard** [krzema12/github-actions-kotlin-dsl#97]
**What feature do you need?**

From https://kotest.io/docs/assertions/soft-assertions.html: 

> Normally, assertions like shouldBe throw an exception when they fail. But sometimes you want to perform multiple assertions in a test, and would like to see all of the assertions that failed. Kotest provides the assertSoftly function for this purpose.

```kotlin
assertSoftly {
  foo shouldBe bar
  foo should contain(baz)
}
```

It would be nice to have something like `assertSoftly`  inside the `workflow()` function
so that we could see all validation errors at once, 
instead of having to do a compilation-run cycle for each one.

**Comment by krzema12** [comment]
Good idea! I'm just after implementing #1 which doesn't produce an exhaustive list of validation errors. This task is a nice follow-up after that.
